docs/docs/repository.mdx

```diff
@@ -26,21 +29,45 @@ The Repository pattern provides 2 main benefits:
 1. Using the pattern, you can replace your data store without changing your business code.
 2. It encourages you to implement all store operations in one place, making your code more reusable and easy to find.
 
-You can also use the `OOP` approach:
+If working in Angular, you can also use the object-oriented programming (OOP) approach:
 
 ```ts title="auth.repository.ts"
-// ...
+import { createStore, select, withProps } from '@ngneat/elf';
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
 
-const authStore = createStore({ name }, withProps<AuthProps>({ user: null }));
+interface AuthProps {
+  user: { id: string } | null;
+}
 
+@Injectable({ providedIn: 'root' })
 export class AuthRepository {
-  user$ = authStore.pipe(select((state) => state.user));
+  user$: Observable<AuthProps['user']>;
+
+  private store;
+
+  constructor() {
+    this.store = this.createStore();
+
+    this.user$ = this.store.pipe(select((state) => state.user));
+  }
+
+  public updateUser(user: AuthProps['user']) {
+    this.store.update((state) => {
+      return {
+        ...state,
+        user,
+      };
+    });
+  }
+
+  private createStore(): typeof store {
```

Comment:
This style will be used only if you set the [inlineStoreInClass](https://ngneat.github.io/elf/docs/cli#config) to `true`.

Reply:
We can add a new block showing this example.

Reply:
You got me thinking.  Should these new examples go in the repository documentation or the CLI documentation.  I could make an argument for both.  What are your thoughts?

Some feedback, the team that I work on is considering using Elf but was very concerned about the lack of Angular examples.  I told them that the CLI produces really great Angular code (in my opinion (IMO)).  Their feedback was that they wished the documentation showed it.

Reply:
We can open a dedicated section of "Angular Examples"

Reply:
Oh .... I like that.  ❤️